src/pages/workspaces/workspace/Data.js

```diff
@@ -563,17 +565,41 @@ const WorkspaceData = _.flow(
     }
   }
 
-  const loadWdsSchema = async () => {
-    if (isFeaturePreviewEnabled('workspace-data-service') && !getConfig().isProd) {
-      try {
-        setWdsSchema([])
-        setWdsSchemaError(undefined)
-        const wdsSchema = await Ajax(signal).WorkspaceData.getSchema(workspaceId)
-        setWdsSchema(wdsSchema)
-      } catch (error) {
-        setWdsSchemaError(error)
-      }
+  const loadWdsSchemaAndUrl = async () => {
+    if (isFeaturePreviewEnabled('workspace-data-service') && !getConfig().isProd && isAzureWorkspace) {
+      await Ajax(signal).Apps.getV2AppInfo(workspaceId).then(async apps => {
+        try {
+          setWdsSchema([])
+          setWdsSchemaError(undefined)
+          const url = getWdsUrl(apps)
+          setWdsProxyUrl(url)
+          const wdsSchema = await Ajax(signal).WorkspaceData.getSchema(url, workspaceId)
+          setWdsSchema(wdsSchema)
+        } catch (error) {
+          setWdsSchemaError(error)
+        }
+      })
+    }
+  }
+
+  const getWdsUrl = apps => {
+    // look explicitly for an app named 'cbas-wds-default'. If found, use it, even if it isn't running
+    // this handles the case where the user has explicitly shut down the app
+    const namedApp = apps.filter(app => app.appType === 'CROMWELL' && app.appName === 'cbas-wds-default')
+    if (namedApp.length === 1) {
+      return namedApp[0].proxyUrls.wds
+    }
+    // if we didn't find the expected app 'cbas-wds-default', go hunting:
+    const candidates = apps.filter(app => app.appType === 'CROMWELL' && app.status === 'RUNNING')
+    if (candidates.length === 0) {
+      // no app deployed yet
+      return ''
+    }
+    if (candidates.length > 1) {
+      // multiple apps found; use the earliest-created one
+      candidates.sort((a, b) => a.auditInfo.createdDate - b.auditInfo.createdDate)
```

Comment:
What's the reason for picking the earliest-created instead of latest-created one?  If earliest seems like it would be impossible to update if that was necessary for some reason.

Reply:
If we picked the latest-created one, then the contents of the data tables would appear to change suddenly - or disappear suddenly - as soon as someone deployed a second app, since the simple presence of the second app would change which app we're looking at.

Picking the earliest-created one is stable (even if that means it is stably pointing at a failed app deployment).

The workaround for the earliest-created app being in a bad state is for the user to call the Leo API to delete (or stop) the earliest app.

Reply:
Gotcha.  Thanks!